.github/workflows/check_task.yml

```
@@ -12,8 +10,82 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
-  build:
+  check-affected-files:
+    name: Check affected files
+    runs-on: ubuntu-latest
+    outputs:
+      affected_file: ${{ steps.check-affected-files.outputs.affected_file }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v45
+
+      - name: Check affected files
+        id: check-affected-files
+        env:
+          # get all affected files (created, modified, deleted, etc.)
+          FILES: ${{ steps.changed-files.outputs.all_changed_and_modified_files }}
+          FILES_COUNT: ${{ steps.changed-files.outputs.all_changed_and_modified_files_count }}
+        run: |
+          for file in "$FILES"; do
+            echo "Info: $file was changed"
+          done
+          if [ "$FILES_COUNT" -ne 1 ]; then
+            echo "Error: More than one file affected ($FILES_COUNT)"
+            exit 1
+          fi
+          if [[ "$FILES" =~ ^contributions\/(executable-tutorial|feedback|open-source)\/[^\/]+\/README\.md$ ]]; then
+            echo "Info: Matches async task"
+          elif [[ "$FILES" =~ ^contributions\/(demo|presentation|scientific-paper)\/week[2-7]\/[^\/]+\/README\.md$ ]]; then
+            echo "Info: Matches sync task"
+          else
+            echo "Error: File in wrong directory ($FILES)"
+            exit 1
+          fi
+
+          echo "affected_file=$FILES" >> "$GITHUB_OUTPUT"
+
+  check-pr-description:
+    name: Check whether Pull Request description matches affected file
+    needs: check-affected-files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Install wdiff
+        # we use wdiff (word-diff) instead of plain diff (line-based) to ensure
+        # that we only enforce *content* to be the same rather than specific
+        # file formatting that would otherwise be ignored when the markdown is
+        # rendered. e.g., wdiff allows the README to break lines at col 80 for
+        # readability while still having the PR description without mid-sentence
+        # breaks (since GitHub descriptions don't support them)
```

What do you mean here? It is unclear. The diff is based on sequence of words, right? 
```suggestion
        # rendered. the comparison takes place between *sequence of words*.
        # eg. "- task 3\n## Category" would be tokenized as ["-task 3", "###Category"]
```
Writing this suggestion, made me question if removes spaces between "-" and "task 3" and so on.

---

Yes, wdiff checks sequence of words, but with a more liberal interpretation of words: anything separated by whitespace is a word, so in your example `- Task 3` is equivalent to `-    Task 3`, but different from `-Task 3` (since then `-Task` would be the word under comparison). This check is fine because `-Task` isn't valid markdown for a list.

---

To be more clear, in the suggestion, "- task 3\n## Category" would be tokenized as ["-", "Task", "3", "##", "Category"] as far as we understand. 

We chose to use wdiff to both workaround the issue of new lines being different in the README and PR description because of Github's automatic formatting of the width as well as to provide students with more  flexibility in formatting with whitespace. At the same time the content is confirmed to be the same and the rendering is probably identical in both cases.

---

That's clear thanks!